File: src/main/java/com/koddy/server/coffeechat/domain/model/response/CoffeeChatDetails.java
package com.koddy.server.coffeechat.domain.model.response;

import com.koddy.server.coffeechat.domain.model.CoffeeChat;
import com.koddy.server.global.utils.encrypt.Encryptor;

import java.time.LocalDateTime;

public record CoffeeChatDetails(
        long id,
        String status,
        String applyReason,
        String suggestReason,
        String cancelReason,
        String rejectReason,
        String question,
        LocalDateTime start,
        LocalDateTime end,
        String chatType,
        String chatValue,
        LocalDateTime createdAt,
        LocalDateTime lastModifiedAt
) {
    public static CoffeeChatDetails of(final CoffeeChat coffeeChat, final Encryptor encryptor) {
        return new CoffeeChatDetails(
                coffeeChat.getId(),
                coffeeChat.getStatus().name(),
                coffeeChat.getReason().getApplyReason(),
                coffeeChat.getReason().getSuggestReason(),
                coffeeChat.getReason().getCancelReason(),
                coffeeChat.getReason().getRejectReason(),
                (coffeeChat.getQuestion() != null) ? coffeeChat.getQuestion() : null,
                (coffeeChat.getReservation() != null) ? coffeeChat.getReservation().getStart() : null,
                (coffeeChat.getReservation() != null) ? coffeeChat.getReservation().getEnd() : null,
                (coffeeChat.getStrategy() != null) ? coffeeChat.getStrategy().getType().getEng() : null,
                (coffeeChat.getStrategy() != null) ? encryptor.decrypt(coffeeChat.getStrategy().getValue()) : null,
                coffeeChat.getCreatedAt(),
                coffeeChat.getLastModifiedAt()
        );
    }
}

File: src/main/java/com/koddy/server/coffeechat/domain/model/response/CoffeeChatDetails.kt
package com.koddy.server.coffeechat.domain.model.response

import com.koddy.server.coffeechat.domain.model.CoffeeChat
import com.koddy.server.global.utils.encrypt.Encryptor
import java.time.LocalDateTime

data class CoffeeChatDetails(
    val id: Long,
    val status: String,
    val applyReason: String?,
    val suggestReason: String?,
    val cancelReason: String?,
    val rejectReason: String?,
    val question: String?,
    val start: LocalDateTime?,
    val end: LocalDateTime?,
    val chatType: String?,
    val chatValue: String?,
    val createdAt: LocalDateTime,
    val lastModifiedAt: LocalDateTime,
) {
    companion object {
        @JvmStatic
        fun of(coffeeChat: CoffeeChat, encryptor: Encryptor): CoffeeChatDetails {
            return CoffeeChatDetails(
                coffeeChat.id,
                coffeeChat.status.name,
                coffeeChat.reason.applyReason,
                coffeeChat.reason.suggestReason,
                coffeeChat.reason.cancelReason,
                coffeeChat.reason.rejectReason,
                coffeeChat.question,
                coffeeChat.reservation?.start,
                coffeeChat.reservation?.end,
                coffeeChat.strategy?.type?.value,
                coffeeChat.strategy?.value?.let { encryptor.decrypt(it) },
                coffeeChat.createdAt,
                coffeeChat.lastModifiedAt,
            )
        }
    }
}
